src/loaders/genome_collection/parse_tool_results.py

```diff
@@ -726,57 +442,50 @@ def microtrait(root_dir, env, kbase_collection, load_ver, fatal_ids):
-
-    # Add _key, collection id and load version to the heatmap metadata and rows
-    heatmap_meta_dict.update({
-        names.FLD_ARANGO_KEY: collection_load_version_key(kbase_collection, load_ver),
-        names.FLD_COLLECTION_ID: kbase_collection,
-        names.FLD_LOAD_VERSION: load_ver
-    })
-
-    heatmap_rows_list = []
-    for r in heatmap_rows:
-        d = r.dict()
-        d.pop(names.FLD_MATCHED, None)  # inserted by the model but not needed in the DB
-        d.pop(names.FLD_SELECTED, None)  # inserted by the model but not needed in the DB
-        heatmap_rows_list.append(dict(
-            # Needs to have the match and selection field inserted
-            d, **init_row_doc(kbase_collection, load_ver, d[names.FLD_KBASE_ID])
-        ))
-    heatmap_cell_details_list = _process_rows_list(heatmap_cell_details_list,
-                                                   kbase_collection,
-                                                   load_ver,
-                                                   FIELD_HEATMAP_CELL_ID,
-                                                   collection_data_id_key)
-
-    return heatmap_meta_dict, heatmap_rows_list, heatmap_cell_details_list
+            # process heatmap cell details
+            with jsonlines.open(data_dir / MICROTRAIT_CELLS, 'r') as jsonl_f:
+                for cell in jsonl_f:
+                    cell[names.FLD_COLLECTION_ID] = kbase_collection
+                    cell[names.FLD_LOAD_VERSION] = load_ver
+                    cell[names.FLD_ARANGO_KEY] = cell[FIELD_HEATMAP_CELL_ID]  # using cell_id (uuid) as _key
+                    heatmap_cell_details.append(cell)
+
+            # process heatmap rows
+            with jsonlines.open(data_dir / MICROTRAIT_DATA, 'r') as jsonl_f:
+                for data in jsonl_f:
+                    # ensure cells are ordered by column id
+                    ordered_cells = sorted(data[FIELD_HEATMAP_ROW_CELLS],
+                                           key=lambda x: int(x[FIELD_HEATMAP_COL_ID]))
+                    for cell in ordered_cells:
+                        cell_val = cell[FIELD_HEATMAP_CELL_VALUE]
+                        min_value = min(min_value, cell_val)
+                        max_value = max(max_value, cell_val)
+                    data[FIELD_HEATMAP_ROW_CELLS] = ordered_cells
+                    heatmap_rows.append(dict(data,
+                                             **init_row_doc(kbase_collection, load_ver, data[names.FLD_KBASE_ID])))
+
+            # process heatmap metadata
+            with jsonlines.open(data_dir / MICROTRAIT_META, 'r') as jsonl_f:
+                metas = [meta for meta in jsonl_f]
+                if reference_meta is None:
+                    reference_meta = metas
+
+                # check if the metadata is consistent across all data dirs
+                if metas != reference_meta:
+                    raise ValueError(f'Inconsistent metadata for {data_dir}')
```

Comment:
There's only one line here for each genome, right? so
```suggestion
                meta = next(jsonl_f)
                if reference_meta is None:
                    reference_meta = meta

                # check if the metadata is consistent across all data dirs
                if meta != reference_meta:
                    raise ValueError(f'Inconsistent metadata for {data_dir}')
```

That confused the heck out of me

Reply:
I suppose you could also read the list, check that `len == 1`, and then `meta = metas[0]`

Reply:
for each genome, we create `microtrait_meta.jsonl` something like:
```
{"col_id": "1", "name": "Aromatic acid transport", "description": "Resource Acquisition:Substrate uptake:aromatic acid transport", "category": "Resource Acquisition", "type": "count"}
{"col_id": "2", "name": "Biopolymer transport", "description": "Resource Acquisition:Substrate uptake:biopolymer transport", "category": "Resource Acquisition", "type": "count"}
{"col_id": "3", "name": "Amino sugar transport", "description": "Resource Acquisition:Substrate uptake:carbohydrate transport:amino sugar transport", "category": "Resource Acquisition", "type": "count"}
...
```
I want to read the meta_data from the first directory. And see if there are any different metadata from the rest dirs.

Reply:
Oh I see, for some reason I thought that it'd look like the metadata model. So 1 line per column, got it.

Could you add a brief comment explaining that?

Reply:
👍 added.

Reply:
I was thinking of something like 

```
# The microtrait runner saves the metadata as jsonl with 1 trait column per line
metas = [meta for meta in jsonl_f]
```

Reply:
👍

Reply:
:+1: